src/Akka.Persistence.Sql.Hosting.Tests/HealthCheckSpec.cs

```diff
@@ -0,0 +1,112 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Hosting;
+using Akka.Hosting.HealthChecks;
+using Akka.Persistence.Journal;
+using Akka.Persistence.Sql.Tests.Common.Containers;
+using FluentAssertions;
+using FluentAssertions.Extensions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Sql.Hosting.Tests
+{
+    /// <summary>
+    /// Validates that health checks are properly registered after the refactoring.
+    /// </summary>
+    public class HealthCheckSpec : Akka.Hosting.TestKit.TestKit, IClassFixture<SqliteContainer>
+    {
+        private readonly SqliteContainer _fixture;
+
+        public HealthCheckSpec(ITestOutputHelper output, SqliteContainer fixture)
+            : base(nameof(HealthCheckSpec), output)
+        {
+            _fixture = fixture;
+
+            if (!_fixture.InitializeDbAsync().Wait(10.Seconds()))
+                throw new Exception("Failed to clean up database in 10 seconds");
+        }
+
+        protected override void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+        {
+            base.ConfigureServices(context, services);
+            services.AddHealthChecks();
+        }
+
+        protected override void ConfigureAkka(AkkaConfigurationBuilder builder, IServiceProvider provider)
+        {
+            // Use the refactored WithSqlPersistence with health check registration
+            builder.WithSqlPersistence(
+                connectionString: _fixture.ConnectionString,
+                providerName: _fixture.ProviderName,
+                journalBuilder: journal =>
+                {
+                    journal.WithHealthCheck(HealthStatus.Degraded);
+                },
+                snapshotBuilder: snapshot =>
+                {
+                    snapshot.WithHealthCheck(HealthStatus.Degraded);
+                });
+        }
+
+        [Fact]
+        public async Task Health_checks_should_be_registered_and_healthy()
+        {
+            // Arrange
+            var healthCheckService = Host.Services.GetRequiredService<HealthCheckService>();
+
+            // Act - run all health checks
+            var healthReport = await healthCheckService.CheckHealthAsync(CancellationToken.None);
+
+            // Assert - verify that health checks are registered and healthy
+            healthReport.Entries.Should().NotBeEmpty("health checks should be registered");
+
+            // Debug: print all registered health checks (ALL of them, not just SQL)
+            Output?.WriteLine($"Total health checks registered: {healthReport.Entries.Count}");
+            foreach (var entry in healthReport.Entries)
+            {
+                Output?.WriteLine($"  - {entry.Key}: {entry.Value.Status}");
+            }
+
+            // We should have exactly 2 health checks: journal and snapshot
```

Comment:
Validate that BOTH health checks (journal + snapshot store) run: this is what helped me find https://github.com/akkadotnet/Akka.Hosting/issues/666